rest/dbHandler.go

```diff
@@ -138,7 +118,73 @@ func (dbHandler DbHandler) deleteTrackPage(w http.ResponseWriter, r *http.Reques
 	return
 }
 
-// Prepares a general response to send back to the client, setting various common variables in the ResponseWriter
+// updatePrivacyPage Changes the Private variable to a new variable
+func (dbHandler DbHandler) updatePrivacyPage(w http.ResponseWriter, r *http.Request) {
+	log := gtbackend.DebugLogPrepareHeader(filenameDBH, "updatePrivacyPage")
+	trackID := r.Header.Get("trackID")
+	newSetting := gtbackend.GetBoolFromString(r.Header.Get("private"))
+	uid := r.Header.Get("uid")
+
+	d, err := UpdatePrivacy(dbHandler.Ctx.App, trackID, uid, newSetting)
+	if err != nil {
+		gtbackend.DebugLogErrNoMsg(log, err)
+
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w = prepareMetadataSingleResponse(w, d, constant.ApplicationJSON)
+	return
+}
+
+// takeOwnershipPage Sets the UID of tracks to this UID if it's the scraper's UID that is taken from
+func (dbHandler DbHandler) takeOwnershipPage(w http.ResponseWriter, r *http.Request) {
+	log := gtbackend.DebugLogPrepareHeader(filenameDBH, "takeOwnershipPage")
+	trackID := r.Header.Get("trackID")
+	uid := r.Header.Get("uid")
+
+	d, err := TakeOwnership(dbHandler.Ctx.App, trackID, uid)
+	if err != nil {
+		gtbackend.DebugLogErrNoMsg(log, err)
+
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w = prepareMetadataSingleResponse(w, d, constant.ApplicationJSON)
+	return
+}
+
+// insertTrackPointPage Inserts TrackPoint data for caching in the database
+func (dbHandler DbHandler) insertTrackPointPage(w http.ResponseWriter, r *http.Request) {
+	log := gtbackend.DebugLogPrepareHeader(filenameDBH, "insertTrackPointPage")
+	trackID := r.Header.Get("trackID")
+	uid := r.Header.Get("uid")
+	rawJSON := r.Header.Get("trackPoints")
+
+	// Parse into objects the string JSON from header
+	var parsed []models.TrackPoint
+	err := json.Unmarshal([]byte(rawJSON), &parsed)
+	if err != nil {
+		gtbackend.DebugLogErrNoMsg(log, err)
+
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
```

Comment:
if err := json.Unmarshal([]byte(rawJSON), &parsed); err != nil {
    .......
}

Reply:
Guess it cuts down on a line.

Reply:
It's a good to know, approving